src/WebJobs.Extensions.DurableTask/Listener/TaskOrchestrationShim.cs

```diff
@@ -80,7 +80,7 @@ public override async Task<string> Execute(OrchestrationContext innerContext, st
                     this.context.IsReplaying));
             }
 
-            object returnValue;
+            object returnValue = null;

```

Comment:
I'm a little concerned that you have to assign this to `null` here. I usually try to make an effort to avoid this to reduce the chances of `NullReferenceExceptions` and to make correctness easier to reason about. I'm wondering if you can refactor the `catch` block a bit to avoid needing to do this. The `TryExtract...` pattern I suggested elsewhere might make this easier.